zebra-consensus/src/block/subsidy/founders_reward.rs

```diff
@@ -58,4 +170,62 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn test_founders_address() -> Result<(), Report> {
+        founders_address_for_network(Network::Mainnet)?;
+        founders_address_for_network(Network::Testnet)?;
+
+        Ok(())
+    }
+
+    fn founders_address_for_network(network: Network) -> Result<(), Report> {
+        // Test if all the founders addresses are paid in the entire period
+        // where rewards are active.
+        let mut populated_addresses: Vec<Address> = Vec::new();
+        let mut hardcoded_addresses = FOUNDERS_REWARD_ADDRESSES_MAINNET;
+
+        if network == Network::Testnet {
+            hardcoded_addresses = FOUNDERS_REWARD_ADDRESSES_TESTNET;
+        }
+
+        // After blossom, founders addresses will change at a greater interval but we should catch
+        // all the changes if we increment by founders_address_change_interval() for
+        // the entire period where founders reward are paid.
+        let mut increment = SLOW_START_SHIFT.0;
+        while founders_reward_active(Height(increment), network) {
+            let founder_address = founders_reward_address(Height(increment), network)?;
+            if !populated_addresses.contains(&founder_address) {
+                populated_addresses.push(founder_address);
+            }
+            increment = increment + founders_address_change_interval().0;
+        }
+
+        //assert_eq!(populated_addresses.len(), hardcoded_addresses.len());
+
+        let mut index = 0;
+        for addr in populated_addresses {
+            assert_eq!(
+                addr,
+                Address::from_str(hardcoded_addresses[index as usize]).expect("an address")
+            );
+            index += 1;
+        }
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_founders_address_count() -> Result<(), Report> {
+        assert_eq!(
+            FOUNDERS_REWARD_ADDRESSES_MAINNET.len() as u32,
+            FOUNDERS_ADDRESS_COUNT
+        );
+        assert_eq!(
+            FOUNDERS_REWARD_ADDRESSES_TESTNET.len() as u32,
```

Comment:
```suggestion
            // The last two addresses are not used, because Canopy activated before
            // the halving on testnet.
            (FOUNDERS_REWARD_ADDRESSES_TESTNET.len() as u32) - 2,
```